packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -433,133 +618,233 @@ fn array_to_user_warnings(array: mv8::Value<'_>) -> Vec<UserWarning> {
     }]
 }
 
-fn get_js_error(_mv8: &MiniV8, return_val: &mv8::Object<'_>) -> Option<Error> {
-    if let Ok(errors) = return_val.get("user_errors") {
-        if !matches!(errors, mv8::Value::Undefined) && !matches!(errors, mv8::Value::Null) {
-            let errors = array_to_user_errors(errors);
+fn get_js_error<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    return_val: v8::Local<'s, v8::Object>,
+) -> Option<Error> {
+    tracing::debug!("Collecting JS errors");
+    let user_errors = match new_js_string(scope, "user_errors") {
+        Ok(user_errors) => user_errors,
+        Err(err) => return Some(err),
+    };
+
+    if let Some(errors) = return_val.get(scope, user_errors.into()) {
+        if !errors.is_null_or_undefined() {
+            let errors = array_to_user_errors(scope, errors);
             if !errors.is_empty() {
                 return Some(Error::User(errors));
             }
         }
     }
 
-    if let Ok(mv8::Value::String(e)) = return_val.get("pkg_error") {
-        // TODO: Don't silently ignore non-string, non-null-or-undefined errors
-        //       (try to convert error value to JSON string and return as error?).
-        return Some(Error::Package(PackageError(e.to_string())));
+    let pkg_error = match new_js_string(scope, "pkg_error") {
+        Ok(pkg_error) => pkg_error,
+        Err(err) => return Some(err),
+    };
+
+    let runner_error = match new_js_string(scope, "runner_error") {
+        Ok(runner_error) => runner_error,
+        Err(err) => return Some(err),
+    };
+
+    if let Some(err) = return_val.get(scope, pkg_error.into()) {
+        // Even though rusty_v8 returns an Option, if the object does not have the property the
+        // result will be Some(undefined)
+        if !err.is_undefined() {
+            let err: v8::Local<'s, v8::String> = if let Some(err) = err.to_string(scope) {
+                err
+            } else {
+                return Some(Error::V8("Could not convert err to String".to_string()));
+            };
+
+            // TODO: Don't silently ignore non-string, non-null-or-undefined errors
+            //       (try to convert error value to JSON string and return as error?).
+            return Some(Error::Package(PackageError(
+                err.to_rust_string_lossy(scope),
+            )));
+        }
     }
 
-    if let Ok(mv8::Value::String(e)) = return_val.get("runner_error") {
-        // TODO: Don't ignore non-string, non-null-or-undefined errors
-        return Some(Error::Embedded(e.to_string()));
+    if let Some(err) = return_val.get(scope, runner_error.into()) {
+        // Even though rusty_v8 returns an Option, if the object does not have the property the
+        // result will be Some(undefined)
```

Comment:
```suggestion
        // result will be `Some(undefined)` rather than `None`
```

(appreciate it's a normal comment and won't be formatted, the backticks are just to help identify None as the Rust thing rather than plain English)